edgekv/src/tests/mod.rs

```diff
@@ -0,0 +1,351 @@
+                i, n_scanned, N,
+            );
+            }
+        }
+    //    t.sync_all().unwrap();
+        //println!("{} ")
+        //drop(t);
+        let t = Arc::new(EdgeKV::open("./testdir/_test_concurrent_tree_ops").unwrap());
+
+        let n_scanned = t.iter().count();
+        if n_scanned != N {
+            warn!(
+                "WARNING: test {} only had {} keys present \
+                 in the DB AFTER restarting. expected {}",
+                i, n_scanned, N,
+            );
+        }
+        //println!("{}", t);
+        debug!("========== reading sets in test {} ==========", i);
+        par! {t, |tree: &EdgeKV, k: Vec<u8>| {
+            if let Some(v) =  tree.get(&k).unwrap() {
+                if v != k {
+                    panic!("expected key {:?} not found", k);
+                }
+                //println!("{:?}",v);
+            } else {
+                panic!("could not read key {:?}, which we \
+                       just wrote to tree {}", k, tree);
+            }
+        }}
+        ;
+
+        drop(t);
+    }
+
+    /*let t = Arc::new(EdgeKV::temp("./testdir/_test_concurrent_tree_ops").unwrap());
+
+    debug!("========== deleting in test {} ==========", i);
+    par! {t, |tree: &EdgeKV, k: Vec<u8>| {
+        tree.delete(&k).unwrap();
+    }};
+
+    drop(t);
+    let t = Arc::new(EdgeKV::temp("./testdir/_test_concurrent_tree_ops").unwrap());
+
+    par! {t, |tree: &EdgeKV, k: Vec<u8>| {
+        if let Ok(None) = tree.get(&k) {
+            assert!(true)
+        }
+        else {
+            assert!(false)
+        }
+        //assert_eq!(tree.get(&k), Ok(None));
+    }};*/
+
+    clean_up("_test_monotonic_inserts");
+}
+
+
+fn concatenate_merge(
+    _key: &[u8],                // the key being merged
+    old_value: Option<Vec<u8>>, // the previous value, if one existed
+    merged_bytes: &[u8],        // the new bytes being merged in
+) -> Option<Vec<u8>> {
+    // set the new value, return None to delete
+    let mut ret = old_value.unwrap_or_else(|| vec![]);
+
+    ret.extend_from_slice(merged_bytes);
+
+    Some(ret)
+}
+
+#[test]
+#[serial]
+fn test_merge_operator() {
+    clean_up("_test_merge_operator");
+    let db = EdgeKV::open("./testdir/_test_merge_operator").unwrap();
+
+    let k = b"k1";
+
+    db.put(k.to_vec(), vec![0]);
+    db.merge(concatenate_merge, k.to_vec(), vec![1]);
+    db.merge(concatenate_merge, k.to_vec(), vec![2]);
+    db.sync_all();
+    assert_eq!(db.get(&k.to_vec()).unwrap().unwrap(), vec![0, 1, 2]);
+
+    // Replace previously merged data. The merge function will not be called.
+    db.put(k.to_vec(), vec![3]);
+    db.sync_all();
+    assert_eq!(db.get(&k.to_vec()).unwrap().unwrap(), vec![3]);
+
+    // Merges on non-present values will cause the merge function to be called
+    // with `old_value == None`. If the merge function returns something (which it
+    // does, in this case) a new value will be inserted.
+    //db.delete(&k.to_vec());
```

Comment:
what about this commented out code in the test? not needed? something that doesn't work now but should work later?